Adding support for macOS

vanshg opened the pull request:
Tries to resolve https://github.com/ksmandersen/GoogleReporter/issues/2 

I haven't been able to test out in an actual macOS application using this library; however, it seems to compile and the alternate implementations of OS specific function seem to produce the desired results. 

Also, some of the project/build settings may need to be modified/fixed as I'm not sure what the correct configuration should be.

vanshg committed:
Adding support for macOS

ksmandersen commented:
Thank you for the PR 💪 !

I have a few comments on your code. 

There is too much code in here for "just" determining the user agent on macOS. This must be simplified in some way. I suggest taking a look at the code in DeviceGuru: https://github.com/InderKumarRathore/DeviceGuru/blob/0a2b8c1e5f4e7df3e20438b762ac091b0206262c/DeviceGuru.swift#L77

Also:
```swift
let size = NSScreen.main()!.frame.size
```
This will crash if there for some reason isn't if there is no main screen available. I don't know if that is ever a posibility, but I would prefer to never have a crashing condition in library code that is non-essential to the workings of the application. Maybe use ``NSScreen.main()?.frame.size ?? .zero`` instead.

vanshg committed:
Greatly simplified hardware model code, and better nil safety

vanshg commented:
Thanks for the feedback. I've updated the branch with simplified code to get the hardware model string, and also implemented better nil safety as you recommended. Let me know if you have anything else you'd like me to change!

ksmandersen commented:
Thanks for the changes 👍  I'll change a few things about the user agent and try to test it out.

I'd be a great help if you could test the library too before i put this in a tagged release.
